### PR TITLE
Alerting: Disable simplified routing when internal alert manager is disabled

### PR DESCRIPTION
**Pull request by soniaAguilarPeiron**
**What is this feature?**
This PR disable simplified routing on alerting UI if Grafana AM is disabled.

**Why do we need this feature?**

Simplified routing only works for internal alertmanager.

**Who is this feature for?**

All users

**Which issue(s) does this PR fix?**:

Fixes #

Please check that:
- [ ] It works as expected from a user's perspective.
- [ ] If this is a pre-GA feature, it is behind a feature toggle.
- [ ] The docs are updated, and if this is a [notable improvement](https://grafana.com/docs/writers-toolkit/writing-guide/contribute-release-notes/#how-to-determine-if-content-belongs-in-a-whats-new-document), it's added to our [What's New](https://grafana.com/docs/writers-toolkit/writing-guide/contribute-release-notes/) doc.

### CODE REVIEW

File: public/app/features/alerting/unified/components/rule-editor/NotificationsStep.tsx [hunk 1]
```diff
@@ -25,6 +26,12 @@ enum RoutingOptions {
   ContactPoint = 'contact point',
 }
 
+function useHasInternalAlertmanagerEnabled() {
+  const { currentData: amChoiceStatus } =
+    alertmanagerApi.endpoints.getGrafanaAlertingConfigurationStatus.useQuery(undefined);
+  return amChoiceStatus?.alertmanagersChoice === 'internal' || amChoiceStatus?.alertmanagersChoice === 'all';
```
Comment (tomratcliffe) [inline public/app/features/alerting/unified/components/rule-editor/NotificationsStep.tsx:32]: Could we use the `AlertmanagerChoice` enum please? 🙏

File: public/app/features/alerting/unified/components/rule-editor/NotificationsStep.tsx [hunk 2]
```diff
@@ -25,6 +26,12 @@ enum RoutingOptions {
   ContactPoint = 'contact point',
 }
 
+function useHasInternalAlertmanagerEnabled() {
+  const { currentData: amChoiceStatus } =
+    alertmanagerApi.endpoints.getGrafanaAlertingConfigurationStatus.useQuery(undefined);
```
Comment (tomratcliffe) [inline public/app/features/alerting/unified/components/rule-editor/NotificationsStep.tsx:31]: Could we destructure the `useQuery` from the API instead please? (just following on from the Slack discussion the other day)

```
const { useGetGrafanaAlertingConfigurationStatusQuery } = alertmanagerApi;
...
```

**Review by tomratcliffe — COMMENTED** [review]
I'd be tempted to have a small loading state for this portion of the app, but I think in practical terms the API is fast, and its far enough down the page that it will have loaded by the time the user gets there. But what do you think?

Would be great to have some tests for this, there's a `setAlertmanagerChoices` mock server config method that should make it easy to check I think?

File: public/app/features/alerting/unified/components/rule-editor/NotificationsStep.tsx (hunk 2 above)
Comment (soniaAguilarPeiron) [reply public/app/features/alerting/unified/components/rule-editor/NotificationsStep.tsx:31]: truee, I forgot it!

**Comment by soniaAguilarPeiron** [comment]
> I'd be tempted to have a small loading state for this portion of the app, but I think in practical terms the API is fast, and its far enough down the page that it will have loaded by the time the user gets there. But what do you think?

We could, but I believe in this case is fast enough?

> Would be great to have some tests for this, there's a setAlertmanagerChoices mock server config method that should make it easy to check I think?
Good idea. let's add a test

**Review by tomratcliffe — APPROVED** [review]

**Pull request merged**